app/src/androidTest/java/ch/epfl/sweng/swenggolf/ShowOfferStatsTest.java

```diff
@@ -0,0 +1,121 @@
+        Config.setUser(user);
+        mActivityRule.launchActivity(new Intent());
+    }
+
+    @Test
+    public void statsAreDisplayedOnOwnOffer() {
+        offer = FilledFakeDatabase.getOffer(0);
+        FragmentTransaction transaction = mActivityRule.getActivity()
+                .getSupportFragmentManager().beginTransaction();
+        transaction.replace(R.id.centralFragment,
+                FragmentConverter.createShowOfferWithOffer(offer))
+                .commit();
+        onView(withId(R.id.show_offer_views)).check(matches(isDisplayed()));
+    }
+
+    @Test
+    public void statsAreHiddenOnNotOwnOffer() {
+        offer = FilledFakeDatabase.getOffer(1);
+        FragmentTransaction transaction = mActivityRule.getActivity()
+                .getSupportFragmentManager().beginTransaction();
+        transaction.replace(R.id.centralFragment,
+                FragmentConverter.createShowOfferWithOffer(offer))
+                .commit();
+        onView(withId(R.id.show_offer_views)).check(matches(not(isDisplayed())));
+    }
+
+    @Test
+    public void statsAreIncremented() {
+        offer = FilledFakeDatabase.getOffer(2);
+
+        //visit offer n°2 with user n°0
+        FragmentTransaction transaction = mActivityRule.getActivity()
+                .getSupportFragmentManager().beginTransaction();
+        transaction.replace(R.id.centralFragment,
+                FragmentConverter.createShowOfferWithOffer(offer))
+                .commit();
+
+        //visit offer n°2 with user n°8 (author)
+        Config.setUser(FilledFakeDatabase.getUser(8));
+        transaction = mActivityRule.getActivity()
+                .getSupportFragmentManager().beginTransaction();
+        transaction.replace(R.id.centralFragment,
+                FragmentConverter.createShowOfferWithOffer(offer))
+                .commit();
+
+        // test if displayed and incremented
+        onView(withId(R.id.show_offer_views)).check(matches(isDisplayed()));
+        ValueListener<Integer> listener = new ValueListener<Integer>() {
+            @Override
+            public void onDataChange(Integer nb) {
+                assert (nb == OfferStats.INITIAL_NB_VIEWS+1); // incremented once
+            }
+
+            @Override
+            public void onCancelled(DbError error) {
+                fail();
+            }
+        };
+
+        OfferStats.getNbViews(listener, offer);
+    }
+
+    @Test
+    public void statsAreDefaultWhenDbNotWorking() {
+        Database.setDebugDatabase(new FakeDatabase(false));
+        offer = FilledFakeDatabase.getOffer(0);
+
+        FragmentTransaction transaction = mActivityRule.getActivity()
+                .getSupportFragmentManager().beginTransaction();
+        transaction.replace(R.id.centralFragment,
+                FragmentConverter.createShowOfferWithOffer(offer))
+                .commit();
+        onView(withId(R.id.show_offer_views)).check(matches(isDisplayed()));
```

Comment:
I don't get what you are testing, the test checks that the stat value takes a default value ? Or that it is displayed by default ?

Reply:
In order to allow the offline use of the app, if it is impossible to recover the number of views from the database, the default value (0 as of now) is displayed so that it is possible to display the Offer nevertheless if it is already loaded.

Reply:
Okay I got it.